doc/source/rllib/package_ref/replay_buffers.rst

```diff
@@ -0,0 +1,132 @@
+.. _replay-buffer-api-reference-docs:
+
+Replay Buffer API
+==================
+
+RLlib comes with a set of extendable replay buffers which are used mostly in Q-Learning algorithms.
+We provide a base :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.ReplayBuffer` class from which you can build your own buffer.
+Some use cases may require :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer`\s.
+You can find buffer types and arguments to modify their behaviour as part of RLlib's default parameters. They are part of
+the ``replay_buffer_config``.
+
+Basic Usage
+===========
+
+You rarely have to instantiate your own replay buffer when running an experiment, but rather configure it as follows.
+The following is from RLlib's examples section and runs the R2D2 algorithm, which runs without `PER <https://arxiv.org/abs/1511.05952>`__ by default, with PER:
+
+.. dropdown:: **Changing a replay buffer configuration**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../../rllib/examples/replay_buffer_api.py
+        :language: python
+        :start-after: __sphinx_doc_basic_replay_buffer_usage__begin__
+        :end-before: __sphinx_doc_basic_replay_buffer_usage_end__
+
+Specifying a buffer type works the same way as specifying an exploration type.
+Here are three ways of specifying a type:
+
+.. dropdown:: **Changing a replay buffer configuration**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../../rllib/examples/documentation/replay_buffer_demo.py
+        :language: python
+        :start-after: __sphinx_doc_replay_buffer_type_specification__begin__
```

Comment:
The example is good, but also needs import statements to work.

An immediate question I had was: how do I "register" my own buffers and how would I address them then? If that's not possible right now, should the non-type-checked 2nd way of doing things be disallowed?

The third way renders weirdly:

```
"ray.rllib.utils.replay_buffers.ReplayBuffer"}  #
    # Specify
    # buffer by path
```

it should also say something less ambiguous than "path" (Python package qualifier or so).

Reply:
At the moment, you don't need to register buffers. Simply passing a class in there is enough.
Passing in pre-instantiated objects will be discussed in the buffers design doc! Changed the docstring.

Reply:
I've also added a "recommended way" comment.